### PR TITLE
Rnn testable doctests

### LINKED ISSUE #2066: Testable docstrings
https://github.com/tensorflow/addons/pull/677 is merged, so we are happy to introduce doctest on the docstring. This feature is awesome that can avoid wrong example usage.

Because the change might be small, please submit one sub-module, for example, activations, per PR. For more details, please refer to https://www.tensorflow.org/community/contribute/docs_ref.

Note that available namespace in our repo is `tf` for `tensorflow`, `np` for `numpy` and `tfa` for `tensorflow_addons`.

- [x] activations: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/activations
- [ ] callbacks: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/callback
- [ ] image: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/image
- [x] layers: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/layers
- [x] losses: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/losses
- [x] metrics: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/metrics
- [ ] optimizers: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/optimizers
- [x] rnn: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/rnn
- [ ] seq2seq: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/seq2seq
- [ ] text: https://github.com/tensorflow/addons/tree/master/tensorflow_addons/text

It's also really appreciated that one can help improve our doc based on https://www.tensorflow.org/community/contribute/docs_style when scrutinizing docstrings.

### PR DESCRIPTION
**Pull request by Harsh188**
# Description

Adding testable docstrings to `rnn`. 

Part of #2066 

Changes made to:
- `esn_cell.py`
- `layer_norm_simple_rnn_cell.py`
- `peephole_lstm_cell.py`

## Type of change

- [x] Updated or additional documentation

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [x] By running pre-commit hooks
- [x] This PR addresses an already submitted issue for TensorFlow Addons

# How Has This Been Tested?

Using pytest

### CODE REVIEW

File: tensorflow_addons/rnn/peephole_lstm_cell.py [hunk 1]
```diff
@@ -39,14 +39,15 @@ class PeepholeLSTMCell(tf.keras.layers.LSTMCell):
...
-    output = layer(input)
-    ```
+    >>> # Create 2 PeepholeLSTMCells
+    >>> peephole_lstm_cells = [PeepholeLSTMCell(size) for size in [128, 256]]
+    >>> # Create a layer composed sequentially of the peephole LSTM cells.
+    >>> layer = tf.keras.layers.RNN(peephole_lstm_cells)
+    >>> input = tf.keras.Input((100, 100))
+    >>> output = tf.keras.layers.Layer(input)
+
+    <... Note: Values used for timesteps and input_dim have been used purely
```
Comment (WindQAQ) [inline tensorflow_addons/rnn/peephole_lstm_cell.py:49]: Is this warning? I think we can at least print output.

Comment (Harsh188) [reply tensorflow_addons/rnn/peephole_lstm_cell.py:49]: I'll include the output of `output` but it's only going to be displayed as a `Layer` object. I think the example tries to depict the way in which the `PeepholeLSTMCells` can be utilized, it doesn't necessarily demonstrate the full process. That's why I included a short note stating that the values used for `input` have no significance.

Comment (WindQAQ) [reply tensorflow_addons/rnn/peephole_lstm_cell.py:51]: Oh I see. Can we change the functional API to
```python
>>> output, state = rnn(input)
>>> output
>>> state
```
This is much more informative.

Comment (Harsh188) [reply tensorflow_addons/rnn/peephole_lstm_cell.py:51]: @WindQAQ Sorry for the late reply I somehow missed this comment. When I try
```python
>>> keras.layers.RNN(peephole_lstm_cells,return_state=True)
>>> layer.states
```
it displays `([None, None], [None, None])`. I'm a little confused on how to view the state of the layer.

**Comment by WindQAQ** [comment]
@Harsh188 We should canonicalize the example. See the following colab for more details. Note that different cell may have different lengths of state, so when `return_state=True`, the values to be unpacked may be different. Also, It would be great that we can have this example for all kind of rnn cell in addons :-)

https://colab.research.google.com/drive/1Ndgy5iseSrcBm6tjQGdnLyMhbszvdAM8?usp=sharing

**Comment by Harsh188** [comment]
@WindQAQ Thank you for providing the colab example! I'll get to work on this and implement them for the other rnn cells as well.

**Comment by Harsh188** [comment]
Friendly ping @WindQAQ :)

**Review by WindQAQ — APPROVED** [review]
Thanks.

**Pull request merged**